src/Integration/Flysystem/S3/src/S3FilesystemV1.php

```diff
@@ -173,22 +173,14 @@ public function has($path)
     {
         $location = $this->applyPathPrefix($path);
 
-        $result = $this->client->getObject(array_merge($this->options, [
+        $result = $this->client->objectExists(array_merge($this->options, [
             'Bucket' => $this->bucket,
             'Key' => $location,
         ]));
 
-        try {
-            $result->resolve();
+        $result->wait();
```

Comment:
`wait` will wait until the object exists.
You should use `isSuccess` or `isFailure` to immediately get if the o ject exists or not.

Reply:
Oops, 
Thank you